### PR TITLE
Metaservice protos

### PR DESCRIPTION
**Pull request by dkoukoul**

### CODE REVIEW

File: lnd/lnrpc/metaservice.proto [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+option go_package = "github.com/pkt-cash/pktd/lnd/lnrpc";
+
+service MetaService {
+    rpc GetInfo2 (GetInfo2Request) returns (GetInfo2Responce);
+}
+
+message GetInfo2Request {}
+
+message neutrino_ban {
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:16]: NeutrinoBan

File: lnd/lnrpc/metaservice.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    string addr = 1;
+    string reason = 2;
+    string endtime = 3;
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:19]: end_time

File: lnd/lnrpc/metaservice.proto [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+message neutrino_query {
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:22]: NeutrinoQuery

File: lnd/lnrpc/metaservice.proto [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	string peer = 1;   
+	string command = 2;        
+	uint32 req_num = 3;  
+	uint32 create_time = 4;      
+	uint32 last_request_time = 5;  
+	uint32 last_response_time = 6;
+}
+
+message neutrino_info {
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:31]: NeutrinoInfo

File: lnd/lnrpc/metaservice.proto [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    peer_desc peers = 1;
+    neutrino_ban bans = 2;
+    neutrino_query queries = 3;
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:34]: Do these three need to be `repeating` ?

File: lnd/lnrpc/metaservice.proto [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    string block_hash = 4;
+    int32 height = 5;
+    string block_timestamp = 6;
+    bool is_syncing = 7;
+}
+
+message wallet_info {
```
Comment (cjdelisle) [inline lnd/lnrpc/metaservice.proto:41]: WalletInfo

File: lnd/lnrpc/pkt.proto [hunk 7]
```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+
+package lnrpc;
+
+option go_package = "github.com/pkt-cash/pktd/lnd/lnrpc";
+
+message DNSSeed {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:7]: Can delete?

File: lnd/lnrpc/pkt.proto [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    string host = 1;
+    bool has_filtering = 2;
+}
+
+message coin_amount {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:12]: Delete, unneeded

File: lnd/lnrpc/pkt.proto [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    string name = 1;
+    string proper_name = 2;
+    int64 units = 3;
+    int32 zeros = 4;
+}
+
+message global_config {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:19]: Delete, unneeded

File: lnd/lnrpc/pkt.proto [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	int64 max_time_offset = 5;
+	int32 median_time_blocks = 6;
+	coin_amount amounts = 7;
+}
+
+message Config {
+    string user_agent_name = 3;
+    string user_agent_version = 4;
+    string user_agent_comments = 5;
+    uint64 services = 7;
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:33]: Make string, stringify the services object and it should print nicely

File: lnd/lnrpc/pkt.proto [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    bool isutc = 4;
+}
+
+message zone {
+    string name = 1;
+    int32 offset = 2;
+    bool isDST = 3;
+}
+
+message Location {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:53]: delete

File: lnd/lnrpc/pkt.proto [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+message zoneTrans {
+    int64 when = 1;
+    uint32 index = 2;
+    bool isstd = 3;
+    bool isutc = 4;
+}
+
+message zone {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:47]: delete

File: lnd/lnrpc/pkt.proto [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+    string user_agent_version = 4;
+    string user_agent_comments = 5;
+    uint64 services = 7;
+    uint32 protocol_version = 8;
+    bool disable_relay_tx = 9;
+    int64 trickle_interval = 11;
+    bool allow_self_conns = 12;
+}
+
+message zoneTrans {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:40]: delete

File: lnd/lnrpc/pkt.proto [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+	string time_connected = 21;
+	int32 starting_height = 22; 
+	int32 last_block = 23;
+	bytes last_announced_block = 24; 
+	uint64 last_ping_nonce = 25;
+	string last_ping_time = 26;
+	int64 last_ping_micros = 27;  
+}
+
+message wallet_stats {
```
Comment (cjdelisle) [inline lnd/lnrpc/pkt.proto:93]: WalletStats

**Comment by dkoukoul** [comment]
I'm happy with it now.

File: lnd/metaservice/service.go [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+	// message before the timeout occurred.
+	ErrUnlockTimeout = er.GenericErrorType.CodeWithDetail("ErrUnlockTimeout",
+		"got no unlock message before timeout")
+)
+
+const (
+	// rpcAuthTimeoutSeconds is the number of seconds a connection to the
+	// RPC server is allowed to stay open without authenticating before it
+	// is closed.
+	rpcAuthTimeoutSeconds = 10
```
Comment (cjdelisle) [inline lnd/metaservice/service.go:29]: What is this ?

Comment (dkoukoul) [reply lnd/metaservice/service.go]: good point, left overs from replicating the service from the walletunlocker.

**Pull request merged**